includes/db-adapter.php

```diff
@@ -93,6 +105,69 @@ private function get_suggested_replacements( $entry ) {
 		return $suggestions;
 	}
 
+	private function morphology_matches( $entry, $suggestion ) {
+		$entry_morphology = $this->rosette_api_morphology( $entry->singular );
```

Comment:
Yeah, since we compare the same original to multiple suggestions, it's better to cache each one separately